plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/storage/TrinoHudiStorage.java

```diff
@@ -0,0 +1,317 @@
+    {
+        return DEFAULT_REPLICATION;
+    }
+
+    @Override
+    public URI getUri()
+    {
+        return URI.create("");
+    }
+
+    @Override
+    public OutputStream create(StoragePath path, boolean overwrite)
+            throws IOException
+    {
+        return fileSystem.newOutputFile(convertToLocation(path)).create();
+    }
+
+    @Override
+    public OutputStream create(
+            StoragePath path,
+            boolean overwrite,
+            Integer bufferSize,
+            Short replication,
+            Long sizeThreshold)
+            throws IOException
+    {
+        return create(path, overwrite);
+    }
+
+    @Override
+    public InputStream open(StoragePath path)
+            throws IOException
+    {
+        return fileSystem.newInputFile(convertToLocation(path)).newStream();
+    }
+
+    @Override
+    public SeekableDataInputStream openSeekable(
+            StoragePath path,
+            int bufferSize,
+            boolean wrapStream)
+            throws IOException
+    {
+        return new TrinoSeekableDataInputStream(
+                fileSystem.newInputFile(convertToLocation(path)).newStream());
+    }
+
+    @Override
+    public OutputStream append(StoragePath path)
+            throws IOException
+    {
+        throw new UnsupportedOperationException("TrinoHudiStorage does not support append operation");
+    }
+
+    @Override
+    public boolean exists(StoragePath path)
+            throws IOException
+    {
+        return fileSystem.newInputFile(convertToLocation(path)).exists();
+    }
+
+    @Override
+    public StoragePathInfo getPathInfo(StoragePath path)
+            throws IOException
+    {
+        Location location = convertToLocation(path);
+        Optional<Boolean> result = fileSystem.directoryExists(location);
+        if (result.isPresent() && result.get()) {
+            return new StoragePathInfo(path, 0, true, (short) 0, 0, 0);
+        }
+        TrinoInputFile inputFile = fileSystem.newInputFile(location);
+        if (!inputFile.exists()) {
+            throw new FileNotFoundException("Path " + path + " does not exist");
+        }
+        return new StoragePathInfo(path, inputFile.length(), false, (short) 0, 0, inputFile.lastModified().toEpochMilli());
+    }
+
+    @Override
+    public boolean createDirectory(StoragePath path)
+            throws IOException
+    {
+        fileSystem.createDirectory(convertToLocation(path));
+        return true;
+    }
+
+    @Override
+    public List<StoragePathInfo> listDirectEntries(StoragePath path)
+            throws IOException
+    {
+        FileIterator fileIterator = fileSystem.listFiles(convertToLocation(path));
+        Set<StoragePathInfo> entryList = new HashSet<>();
+        while (fileIterator.hasNext()) {
+            entryList.add(getDirectEntryPathInfo(path, fileIterator.next()));
+        }
+        if (entryList.isEmpty()) {
+            throw new FileNotFoundException("Path " + path + " does not exist");
```

Comment:
We should use TrinoException if it's user-facing exception. Same for listDirectEntries.

Reply:
Sounds good.  I've changed all the `FileNotFoundException` usage to `TrinoFileSystemException`.

Reply:
Addressed.